### PR TITLE
Initial check-in for roscpp_azure_iothub sample

### PR DESCRIPTION
**Pull request by seanyen**
This is a refresh to ros_azure_iothub sample code. This change is adding an roscpp version of Azure IoT Hub Relay node which is using Azure IoT C99 SDK.

Also, this change fixes some minor issues of the rospy_azure_iothub:
* Add missing exec_depend for rosbridge_libarary.
* Add missing install for sample.launch.

### CODE REVIEW

File: rospy_azure_iothub/package.xml [hunk 1]
```diff
@@ -14,8 +14,8 @@
   <build_depend>rospy</build_depend>
   <build_depend>std_msgs</build_depend>
 
-  <exec_depend>rospy</exec_depend>
   <exec_depend>rospy</exec_depend>
   <exec_depend>std_msgs</exec_depend>
-  <exec_depend>azure-iothub-device-client-pip</exec_depend>
+  <exec_depend>python-azure-iothub-device-client</exec_depend>
```
Comment (seanyen) [inline rospy_azure_iothub/package.xml:19]: I renamed `azure-iothub-device-client-pip` to `python-azure-iothub-device-client` to align with the naming convention we saw in rosdistro database.

**Comment by seanyen** [comment]
@anthturner Not sure if we should add `melodic-devel` branch for this merge? (that's to reflect the fact we are only done testing on melodic for this new addition.)

**Comment by ooeygui** [comment]
Ros nodes are not typically prefixed with roscpp. Suggest dropping that from the prefix.

**Comment by anthturner** [comment]
Let's make sure we're following the guide below... @seanyen-msft and @ooeygui can you verify our naming conventions are in line with the ROS standards?

http://www.ros.org/reps/rep-0144.html

**Pull request merged**